From e5a0219ac0190ec7ed472f515987bcfaee2cfb9c Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Thu, 27 Jan 2011 15:02:30 +0800
Subject: [PATCH] ENGR00118558 MX51/MX53: change the boot command of android boards.

The android uImage is greater than 3M, so we need to change the
boot command.

Signed-off-by: Zhang Jiejing <jiejing.zhang@freescale.com>
---
 include/configs/mx51_bbg_android.h |    8 ++++----
 include/configs/mx53_evk_android.h |    8 ++++----
 include/configs/mx53_smd_android.h |    8 ++++++--
 3 files changed, 14 insertions(+), 10 deletions(-)

diff --git a/include/configs/mx51_bbg_android.h b/include/configs/mx51_bbg_android.h
index 4b1271a..04607a1 100644
--- a/include/configs/mx51_bbg_android.h
+++ b/include/configs/mx51_bbg_android.h
@@ -127,7 +127,7 @@
 	"rootfs=ext4 di1_primary"
 #define CONFIG_ANDROID_RECOVERY_BOOTCMD_MMC  \
 	"run bootargs_base bootargs_android_recovery;"	\
-	"mmc read 0 ${loadaddr} 0x800 0x1800;bootm"
+	"mmc read 0 ${loadaddr} 0x800 0x2000;bootm"
 #define CONFIG_ANDROID_RECOVERY_CMD_FILE "/recovery/command"
 
 #define CONFIG_ANDROID_SYSTEM_PARTITION_MMC 2
@@ -182,15 +182,15 @@
 			"setenv filesize; saveenv\0"			\
 		"bootcmd=run bootcmd_SD \0"				\
 		"bootcmd_SD=run bootargs_base bootargs_android;"	\
-		     "mmc read 0 ${loadaddr} 0x800 1800;"		\
-		     "mmc read 0 ${rd_loadaddr} 0x2000 0x258;"		\
+		     "mmc read 0 ${loadaddr} 0x800 2000;"		\
+		     "mmc read 0 ${rd_loadaddr} 0x3000 0x258;"		\
 		     "bootm ${loadaddr} ${rd_loadaddr}\0"		\
 		"bootargs_android=setenv bootargs ${bootargs}  "	\
 		     "androidboot.console=ttymxc0 init=/init "		\
 		     "di1_primary calibration\0"			\
 		"bootcmd_android_recovery=run bootargs_base"		\
 		     " bootargs_android_recovery;"			\
-		     "mmc read 0 ${loadaddr} 0x800 0x1800;bootm\0"	\
+		     "mmc read 0 ${loadaddr} 0x800 0x2000;bootm\0"	\
 		"bootargs_android_recovery=setenv bootargs ${bootargs}" \
 		     " init=/init root=/dev/mmcblk0p4 rootfs=ext4"	\
 		     " di1_primary \0"					\
diff --git a/include/configs/mx53_evk_android.h b/include/configs/mx53_evk_android.h
index 00007e7..4552cd1 100644
--- a/include/configs/mx53_evk_android.h
+++ b/include/configs/mx53_evk_android.h
@@ -103,7 +103,7 @@
 #define CONFIG_ANDROID_RECOVERY_BOOTARGS_MMC \
 	"setenv bootargs ${bootargs} init=/init root=/dev/mmcblk0p4 rootfs=ext4"
 #define CONFIG_ANDROID_RECOVERY_BOOTCMD_MMC  \
-	"run bootargs_base bootargs_android_recovery;mmc read 0 ${loadaddr} 0x800 0x1800;bootm"
+	"run bootargs_base bootargs_android_recovery;mmc read 0 ${loadaddr} 0x800 0x2000;bootm"
 #define CONFIG_ANDROID_RECOVERY_CMD_FILE "/recovery/command"
 
 #define CONFIG_ANDROID_SYSTEM_PARTITION_MMC 2
@@ -176,14 +176,14 @@
 		     "video=mxcdi0fb:RGB565,800x480M@55 calibration\0"	\
 		"bootcmd=run bootcmd_SD \0"				\
 		"bootcmd_SD=run bootargs_base bootargs_android;"	\
-		     "mmc read 0 ${loadaddr} 0x800 1800;"		\
-		     "mmc read 0 ${rd_loadaddr} 0x2000 0x258;"		\
+		     "mmc read 0 ${loadaddr} 0x800 2000;"		\
+		     "mmc read 0 ${rd_loadaddr} 0x3000 0x258;"		\
 		     "bootm ${loadaddr} ${rd_loadaddr}\0"		\
 		"bootcmd_net=run bootargs_base bootargs_nfs; "		\
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
 		"bootcmd_android_recovery=run bootargs_base"		\
 		     " bootargs_android_recovery;"			\
-		     "mmc read 0 ${loadaddr} 0x800 0x1800;bootm\0"	\
+		     "mmc read 0 ${loadaddr} 0x800 0x2000;bootm\0"	\
 		"bootargs_android_recovery=setenv bootargs ${bootargs}" \
 		     " init=/init root=/dev/mmcblk0p4 rootfs=ext4\0"	\
 
diff --git a/include/configs/mx53_smd_android.h b/include/configs/mx53_smd_android.h
index dcf71c1..953ceba 100644
--- a/include/configs/mx53_smd_android.h
+++ b/include/configs/mx53_smd_android.h
@@ -168,8 +168,12 @@
 		     "video=mxcdi1fb:RGB666,XGA ldb gpu_memory=32M\0"   \
 		"bootcmd=run bootcmd_SD \0"                             \
 		"bootcmd_SD=run bootargs_base bootargs_android;"        \
-		     "mmc read 0 ${loadaddr} 0x800 1800;"               \
-		     "mmc read 0 ${rd_loadaddr} 0x2000 0x258;"          \
+		     "mmc read 0 ${loadaddr} 0x800 2000;"               \
+		     "mmc read 0 ${rd_loadaddr} 0x3000 0x258;"          \
+		     "bootm ${loadaddr} ${rd_loadaddr}\0"               \
+		"bootcmd_eMMC=run bootargs_base bootargs_android;"        \
+		     "mmc read 1 ${loadaddr} 0x800 2000;"               \
+		     "mmc read 1 ${rd_loadaddr} 0x3000 0x258;"          \
 		     "bootm ${loadaddr} ${rd_loadaddr}\0"               \
 		"bootcmd_net=run bootargs_base bootargs_nfs; "          \
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"       \
-- 
1.5.4.4

